src/Shared/TaskLoggingHelper.cs

```diff
@@ -1016,6 +1016,28 @@ public void LogWarning
             // that gives the user something.
             bool fillInLocation = (String.IsNullOrEmpty(file) && (lineNumber == 0) && (columnNumber == 0));
 
+            // This warning will be converted to an error if:
+            // 1. Its code exists within WarningsAsErrors
+            // 2. If WarningsAsErrors is a non-null empty set (treat all warnings as errors)
+            if (BuildEngine is IBuildEngine8 be8 && be8.WarningsAsErrors != null && (be8.WarningsAsErrors.Contains(warningCode) || be8.WarningsAsErrors.Count == 0))
```

Comment:
I think it'll be better if we make this contract explicit. An empty collection for all warnings as errors is confusing. 

Do we already have WarningsAsErrors somewhere? If not, we could use a trick and return an implementation of `ICollection<string>` that always returns `true` from the `Contains()` method.